packages/bus-query-operation/lib/Bindings.ts

```diff
@@ -148,29 +153,93 @@ export function materializeOperation(
         };
       }
 
-      const variables = op.variables.filter(variable => !bindings.has(variable));
+      // Only include non-projected variables in the sub-bindings that will be passed down recursively.
+      // This will result in non-projected variables being replaced with their InitialBindings values.
+      let subBindings = bindingsFactory.fromBindings(bindings);
```

Comment:
> Why do you create a copy of `bindings`? Since bindings are immutable, I think this is not needed here.

In response to your other comment I removed this variable alltogether and work with the bindings object. Otherwise I would say the copying of bindings is necessary in order to instantiate the variable you mention in this comment.